### PR TITLE
Run mypy on Windows and Linux

### PR DESCRIPTION
**Pull request by crusaderky**
Run mypy for Windows-specific issues.
Align pre-commit config to dask/distributed.
Twin of https://github.com/dask/distributed/pull/7094

### CODE REVIEW

File: dask/array/tests/test_testing.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from dask.array.utils import assert_eq
 
 
-@pytest.mark.skipif(sys.flags.optimize, reason="Assertions disabled.")
+@pytest.mark.skipif(bool(sys.flags.optimize), reason="Assertions disabled.")
```
Comment (jrbourbeau) [inline dask/array/tests/test_testing.py:10]: Running locally I get `sys.flags.optimize = 0`. When I specify the `-O`/ `-OO` options when running `python` I get `1` and `2`, respectively. Are those all the values this flag can be?

**Review by ian-r-rose — APPROVED** [review]
Thanks @crusaderky, looks reasonable to me

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -5,55 +5,85 @@ repos:
...
+    hooks:
       - id: flake8
         language_version: python3
         additional_dependencies:
-          - flake8-bugbear==22.8.23  # NOTE: manually update this
-  -   repo: https://github.com/codespell-project/codespell
-      rev: v2.1.0
-      hooks:
+          # NOTE: autoupdate does not pick up flake8-bugbear since it is a transitive
+          #  dependency. Make sure to update flake8-bugbear manually on a regular basis.
```
[the remark this answers is not in the document]
Comment (jrbourbeau) [reply .pre-commit-config.yaml:37]: Pushed a commit up for this

**Review by jrbourbeau — APPROVED** [review]
Thanks @crusaderky -- will merge after CI finishes

File: dask/array/tests/test_testing.py (hunk 1 above)
Comment (crusaderky) [reply dask/array/tests/test_testing.py:10]: Yes. In the tuture there may be >2 (unlikely, but possible). But 0 will continue meaning 0.

**Pull request merged**